src/components/step-sequence/step-sequence-item/step-sequence-item.js

```diff
@@ -0,0 +1,77 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Icon from '../../icon';
+import './step-sequence-item.scss';
+
+const baseClass = 'carbon-step-sequence-item';
+const classes = status => `${baseClass} ${baseClass}--${status}`;
+
+const stepMarker = (status, indicator) => {
+  return status === 'complete' ? <Icon type='tick' /> : indicator;
+};
+
+const ariaLabelProp = (ariaLabel) => {
+  return ariaLabel ? { 'aria-label': ariaLabel } : {};
+};
+
+const ariaRoleProp = (status) => {
+  return status === 'current' ? { 'aria-current': 'step' } : {};
+};
+
+const completeLabel = (label, status) => {
+  if (label && status === 'complete') {
+    return <span className='carbon-step-sequence-item__visually-hidden'>{ label }</span>;
+  }
+
+  return null;
+};
+
+const currentLabel = (label, status) => {
+  if (label && status === 'current') {
+    return <span className='carbon-step-sequence-item__visually-hidden'>{ label }</span>;
+  }
+
+  return null;
+};
+
+const StepSequenceItem = ({
+  ariaLabel,
+  children,
+  indicator,
+  status,
+  hiddenCompleteLabel,
+  hiddenCurrentLabel
+}) => (
+  <li
+    className={ classes(status) }
+    { ...ariaLabelProp(ariaLabel) }
```

Comment:
Maybe we should just let developers add any additional props they want rather than adding specific support, otherwise we need to support/maintain these props. Eg.

```
const StepSequenceItem = ({
  children,
  indicator,
  status,
  hiddenCompleteLabel,
  hiddenCurrentLabel,
  ...props
}) => (
  <li
    { ...props }
    className={ classes(status) }
    { ...ariaRoleProp(status) }
  >
```